### PR TITLE
Restructure minor optics computation

### PR DESCRIPTION
**Pull request by sriharshakandala**
## Purpose 
Restructure minor optics computation

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/optics/GasOptics.jl [hunk 1]
```diff
@@ -224,54 +252,39 @@ Compute optical thickness contributions from minor gases.
...
-        kminor = lkp.kminor_lower
-    else # in upper atmosphere
-        minor_bnd_st = lkp.minor_upper_bnd_st
-        idx_gases_minor = lkp.idx_gases_minor_upper
-        minor_scales_with_density = lkp.minor_upper_scales_with_density
-        idx_scaling_gas = lkp.idx_scaling_gas_upper
-        scale_by_complement = lkp.upper_scale_by_complement
-        minor_gpt_sh = lkp.minor_upper_gpt_sh
-        kminor = lkp.kminor_upper
-    end
```
Comment (szy21) [inline src/optics/GasOptics.jl:245]: Why don't we need to check tropo in this function anymore?

Comment (sriharshakandala) [reply src/optics/GasOptics.jl:245]: We do the check here: https://github.com/CliMA/RRTMGP.jl/pull/433/files#diff-2d88c4dcffc62c244ce50270af42067f76971a8a3e2edd44c4100a87f6ee8bcfR190 and pass in the appropriate minor optics lookup table to this function

File: src/optics/GasOptics.jl [hunk 2]
```diff
@@ -224,61 +222,45 @@ Compute optical thickness contributions from minor gases.
...
+                        scaling *= density_fact
+                        if idx_scaling_gas > 0
+                            if scale_by_complement == 1
+                                scaling *= (FT(1) - get_vmr(vmr, idx_scaling_gas, glay, gcol) * dry_fact)
+                            else
+                                scaling *= get_vmr(vmr, idx_scaling_gas, glay, gcol) * dry_fact
+                            end
+                        end
                     end
+                    τ_minor += interp2d(fη1, fη2, ftemp, view(kminor, :, :, st_gpt + i), jη1, jη2, jtemp) * scaling
```
Comment (szy21) [inline src/optics/GasOptics.jl:253]: Could you just double check the index is correct? I'm not very familiar with the lookup table structure.

File: src/optics/LookUpTables.jl [hunk 3]
```diff
@@ -687,21 +632,43 @@ function LookUpSW(ds, ::Type{FT}, ::Type{DA}) where {FT <: AbstractFloat, DA}
             minor_upper_bnd_st[ibnd] = UI8(loc_upp + 1)
         end
     end
+    # reorder kminor
```
Comment (szy21) [inline src/optics/LookUpTables.jl:635]: What is this reorder doing?

**Review by szy21 — APPROVED** [review]
I'm not very familiar with the lookup table structure, but the code structure change looks good to me.

File: src/optics/LookUpTables.jl (hunk 3 above)
Comment (sriharshakandala) [reply src/optics/LookUpTables.jl:635]: It places data corresponding to each g-point contiguously

**Pull request merged**